Makefile

```diff
@@ -1,3 +1,5 @@
+GOOS ?= $(shell go env GOOS)
```

Comment:
we  currently build for multiple OS/arch by executing `make cli-images` - does that still work with this change?

Reply:
Ahhhh that's why all of the targets are there, now I understand. I am afraid by updating target `clis` to `dist/argo-$(GOOS)-$(GOARCH)`, only the image fit local machine architecture will be built, but target `dist/argoexec-$(GOOS)-$(GOARCH)` works well. 

I am gonna think of a method to fix this problem. If you have any suggestions also do let me know please :)))

Reply:
maybe inline rather than use patterns :(

Reply:
Hi @alexec ! I have updated code, and personally I think this is the best way of achieve both adding arm support and at the same time build all supported platform images without harming the code readability and method of using this Makefile. Please review again. Thank you very much!

Reply:
It is very weird that the Golang CI failed at `Error: failed to run command "make vendor server/static/files.go CI=true": exit status 2` :( even if there is no vendor in current Makefile...

Reply:
ignore GolangCD - it is busted